semantics/language/execution/expr/function-call.k

```diff
@@ -128,17 +127,13 @@ module C-EXPR-FUNCTION-CALL
           ) ...</call-stack>
           <control>
                <curr-tu> OldTu:K => Tu </curr-tu>
-               <curr-function> CurrFun:K </curr-function>
+               <curr-function> CurrFun:K => X </curr-function>
```

Comment:
I'm not really sure what the best solution is, but probably for now the current code location should just be invalidated at the time of a call. Or maybe set to some special value to indicate that we're currently binding params for a call to function x from location blah. Or maybe the error reporting mechanism could be made smarter in order to detect this case.

Setting curr-function like that might keep it consistent with the current code loc, but it's inconsistent with the current TU and the current call stack (also it breaks some restrict checks).